### PR TITLE
Rename worker_spec attribute

### PR DESCRIPTION
**Pull request by ocaisa**
Short-term fix for #280

For tests to pass, requires #284

### CODE REVIEW

**Comment by ocaisa** [comment]
Closing and  reopening to see if that affects tests passing

**Comment by danpf** [comment]
thank you for doing this @ocaisa !!!

one of your problems is that ncores is now nthreads

I'm not really sure what the other problems could be. some of the complaints are about AttributeErrors but that should be caught by:
https://github.com/dask/distributed/blob/master/distributed/deploy/adaptive.py#L216
(and the line numbers in the error match so it should be okay?)

**Comment by danpf** [comment]
you can revert that last change, requirements.txt isn't respected :(
it's built by conda and just uses the newest (in this case 2.0)
https://github.com/dask/dask-jobqueue/blob/master/ci/slurm/Dockerfile#L8

I would suggest giving `JobQueuePlugin` a `workers` property that returns `self.scheduler.workers`

I don't actually know if that is the answer, but digging deeper into the logs 
```
Traceback (most recent call last):
  File "/opt/anaconda/lib/python3.6/site-packages/tornado/ioloop.py", line 743, in _run_callback
    ret = callback()
  File "/opt/anaconda/lib/python3.6/site-packages/tornado/ioloop.py", line 767, in _discard_future_result
    future.result()
  File "/opt/anaconda/lib/python3.6/site-packages/tornado/gen.py", line 209, in wrapper
    yielded = next(result)
  File "/opt/anaconda/lib/python3.6/site-packages/distributed/deploy/adaptive.py", line 259, in _adapt
    recommendations = self.recommendations()
  File "/opt/anaconda/lib/python3.6/site-packages/distributed/deploy/adaptive.py", line 218, in recommendations
    current = len(self.cluster.workers)
AttributeError: 'SLURMCluster' object has no attribute 'workers'
tornado.application - ERROR - Exception in callback functools.partial(<bound method IOLoop._discard_future_result of <tornado.platform.asyncio.AsyncIOLoop object at 0x7f81de0b95f8>>, <Future finished exception=AttributeError("'SLURMCluster' object has no attribute 'workers'",)>)
Traceback (most recent call last):
  File "/opt/anaconda/lib/python3.6/site-packages/distributed/deploy/adaptive.py", line 216, in recommendations
    current = len(self.cluster.worker_spec)
AttributeError: 'SLURMCluster' object has no attribute 'worker_spec'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/opt/anaconda/lib/python3.6/site-packages/tornado/ioloop.py", line 743, in _run_callback
    ret = callback()
  File "/opt/anaconda/lib/python3.6/site-packages/tornado/ioloop.py", line 767, in _discard_future_result
    future.result()
  File "/opt/anaconda/lib/python3.6/site-packages/tornado/gen.py", line 209, in wrapper
    yielded = next(result)
  File "/opt/anaconda/lib/python3.6/site-packages/distributed/deploy/adaptive.py", line 259, in _adapt
    recommendations = self.recommendations()
  File "/opt/anaconda/lib/python3.6/site-packages/distributed/deploy/adaptive.py", line 218, in recommendations
    current = len(self.cluster.workers)
AttributeError: 'SLURMCluster' object has no attribute 'workers'
```

that's probably at least one of the problems

if you have a better idea feel free to try that instead, i think that should get us in the right direction though

**Comment by ocaisa** [comment]
Thanks @danpf , that suggestion worked. The last issue is that when using Python2, we still have`ncores` rather than `nthreads`, I can make this work with an if/else to get this set of tests to pass but it's probably an indicator of a necessary break with Python2 support.

**Comment by willirath** [comment]
Dask has dropped Python 2 with v2.0.0:

- https://distributed.dask.org/en/latest/changelog.html#id2
- https://docs.dask.org/en/latest/changelog.html#core

In my opinion, this should be reflected here (the repo) as well.

**Comment by guillaumeeb** [comment]
I agree with @willirath on the Python 2 support.

I'll try to review the PR next week, but if someone else (@danpf @willirath) can do it before, I'll be quite happy!

File: dask_jobqueue/tests/test_sge.py [hunk 1]
```diff
@@ -32,7 +32,12 @@ def test_basic(loop):
             workers = list(client.scheduler_info()["workers"].values())
             w = workers[0]
             assert w["memory_limit"] == 2e9 / 4
-            assert w["ncores"] == 2
+            if "nthreads" in w:
+                nthreads = "nthreads"
+            else:
+                # for python2
+                nthreads = "ncores"
+            assert w[nthreads] == 2
```
Comment (danpf) [inline dask_jobqueue/tests/test_sge.py]: revert changes for ncores, we will remove python2 from testing

**Comment by ocaisa** [comment]
I looked at what was done in `distributed` (which is what you suggested) and, given the behaviour we've seen in the testing here, it looks like that  is enough to stop serving to python2.

**Comment by danpf** [comment]
Great! thanks @ocaisa & nice job! everything looks good to me!

**Comment by willirath** [comment]
Looks good to me as well.  The pedant in me tells me that this should be two PR's:

1. Drop Python 2 testing.  Should also add a clear statement about dropping 2 to the https://github.com/dask/dask-jobqueue/blob/master/docs/source/changelog.rst

2. Fix #280

**Comment by ocaisa** [comment]
@willirath I agree you with there, something like dropping python2 support deserves a distinct PR. Will open now.

File: .travis.yml [hunk 2]
```diff
@@ -9,19 +9,10 @@ services:
   - docker
 matrix:
   include:
-    - python: "2.7"
```
Comment (willirath) [inline .travis.yml]: Don't do this here, but tackle in #283.

File: .travis.yml [hunk 3]
```diff
@@ -9,19 +9,10 @@ services:
...
-      env:
-        - OS=ubuntu-14.04
-        - JOBQUEUE=sge
     - python: "3.6"
       env:
         - OS=ubuntu-14.04
         - JOBQUEUE=sge
-    - python: "2.7"
```
Comment (willirath) [inline .travis.yml]: Don't do this here, but tackle in #283.

File: setup.py [hunk 4]
```diff
@@ -21,6 +21,7 @@
     description="Easy deployment of Dask Distributed on job queuing systems "
     "such as PBS, Slurm, or SGE.*",
     url="https://github.com/dask/dask-jobqueue",
+    python_requires=">3.5.0",
```
Comment (willirath) [inline setup.py]: Don't do this here, but tackle in #283.

**Comment by willirath** [comment]
@ocaisa Sorry again for the confusion.  The proposed separation was not so much about the importance or impact of dropping Python 2 warranting a separate PR, but about keeping a clean history that separates commits to master when different purposes are served.  I have tried to clarify this in the reviews of #282 and #283 now, I hope:

#282 just fixes `worker_spec` and disregards red marks on Py2 related tests for now.

#283 drops Python 2 testing but doesn't touch anything else.

**Pull request merged**